### ISSUE TITLE
Simutaneous removal of Safezone/CotH and Royal Decree Bug

### ISSUE THREAD

**Issue by Brisingr14** [Fluorohydride/ygopro#848]
The games handling of this situation is currently not correct. I'll pose an example to iterate this.

1) Safe Zone attached to a face up Cosmic Fortress Gol'Gar
2) You Activate Royal Decree
3) You activate Cosmic Fortress Gol'Gar to bounce back your Decree and Safe Zone

What SHOULD happen at this point is that Gol'gar should remain on the field while both Safezone and Royal Decree return to the hand. In Devpro (and YGOPro), Cosmic Fortress Gol'gar is destroyed.

There are several interpretations why this should be the case, but I like this explanation the best:

If a continuous trap is being negated while on the field, it doesn't remember its target when it leaves. That's why Call of the Haunted and Jinzo make for a [very well known] combo. Jinzo doesn't negate traps that aren't on the field, yet it isn't destroyed when Call of the Haunted leaves play. Safe Zone is basically the same as Call of the Haunted with its "when this leaves play, destroy the monster" ability. If it's negated up to the point of its removal, the effect that's supposed to attach it to a creature isn't applied on the field, so nothing should be removed when it's off the field.

Basically if CotH or Safezone is being negated, it forgets its target. The effect of destroying the monster happens at the same time its removed, however if its negated, it can't destroy its target. Since Royal Decree's is a continuous trap card, it only stops applying its effect when it leaves. Hence a simultaneous removal would result in the same result as if CotH or Safezone was just being negated.

I hope I have explained this sufficiently enough. Please look into this.
